llmfoundry/callbacks/hf_checkpointer.py

```diff
@@ -457,15 +475,24 @@ def run_event(self, event: Event, state: State, logger: Logger) -> None:
             if self._any_register_processes_error(
                 state.device,
             ) and self.final_register_only:
-                log.error(
-                    'An error occurred in one or more registration processes. Fallback to saving the HuggingFace checkpoint.',
-                )
-                self._save_checkpoint(
-                    state,
-                    logger,
-                    upload_to_save_folder=True,
-                    register_to_mlflow=False,
-                )
+                if self.save_folder:
+                    log.error(
+                        'An error occurred in one or more registration processes. Fallback to saving the HuggingFace checkpoint.',
+                    )
+                    self._save_checkpoint(
+                        state,
+                        logger,
+                        upload_to_save_folder=True,
```

Comment:
don't we want the fallback to be to log to mlflow without registering if save_folder is none?

Reply:
Hm we could do that. I was thinking to avoid the log_model path completely, but that could work

Reply:
Actually, assuming that logging happens prior to registration:

If the run errors during registration, then the model was probably already logged. If it errors during logging, we probably don't want to do it again.

Reply:
resolved offline, will keep as is bc the model will already be logged if there are registration failures